pkg/maintenance/postgresql.go

```diff
@@ -102,15 +120,29 @@ func (p *PostgreSQL) DoMaintenance(ctx context.Context) error {
 	}
 
 	p.log.Info("Repacking databases...")
-	err = p.createRepack(clusterName)
+	err = p.createRepack(sgCluster.GetName())
 	if err != nil {
 		return fmt.Errorf("cannot create repack: %v", err)
 	}
 
 	return nil
 }
 
-func (p *PostgreSQL) upgradeVersion(currentVersion string, clusterName string) (OpName, error) {
+func (p *PostgreSQL) checkRestart(sgCluster stackgresv1.SGCluster) (bool, error) {
+	if sgCluster.Status.Conditions == nil {
+		return false, nil
+	}
+	sgClusterConditions := *sgCluster.Status.Conditions
+	for _, condition := range sgClusterConditions {
+		if *condition.Reason == "ClusterRequiresUpgrade" && *condition.Status == "True" {
+			p.log.Info("Doing a security maintenance")
+			return true, p.createSecurityUpgrade(sgCluster.GetName())
```

Comment:
I would get this function out of `checkRestart()` as the name of the method is miss leading since you are doing a security upgrade in the happy path.

Reply:
Taken it out of the function